Lib/test/test_zipfile.py

```diff
@@ -548,6 +548,18 @@ def test_add_file_before_1980(self):
         os.utime(TESTFN, (0, 0))
         with zipfile.ZipFile(TESTFN2, "w") as zipfp:
             self.assertRaises(ValueError, zipfp.write, TESTFN)
+            zipfp.write(TESTFN, strict_timestamps=False)
```

Comment:
I would prefer to reopen the file for the second test, since it's hard to guess what happens when on error occurs. I prefer to limit side effets of unit tests.